docs/data-storage.md

```diff
@@ -0,0 +1,567 @@
+
+The seed state is stored as a flat structure, serialized and stored.
+
+Each set of bakers is stored
+
+### Cryptographic parameters
+
+These are stored by serializing the structure. There are currently no updates to
+this value, so only a single copy is ever stored, the one given in genesis.
+
+### Chain parameter and protocol updates
+
+This part of the state stores
+- current keys authorized to sign the different update types
+- whether a protocol update is currently in effect. This is used to stop
+  consensus.
+- current values of chain parameters, this are reward fractions, inflation rate, etc.
+- pending updates
+
+The **pending updates** consist of an update queue for each update type. All
+**update queues** follow the same overall structure and differ only in the exact
+payload that they store. This payload is dependent on the exact update type.
+
+Each **update queue** consist of the next sequence number for that update, as
+well as a list of pairs of effective time and the payload. The payload is stored
+in a way that allows sharing. Enqueueing a new update will not copy the payload,
+but it will make a copy of the rest of the update queue.
+
+### Release schedule
+
+The release schedule is a result of scheduled transfers. The main complication
+is that the effect of the transaction is not only immediate, but the GTU are
+release over a period of time. This part of the state keeps the information
+needed to efficiently check whether any releases need to be affected at the
+beginning of each block. In particular this part just maps account addresses to
+the timestamp of the next release. This is used together with the information
+stored for each account in the account table, which contains all remaining
+releases.
+
+Since this part of the state is uniquely determined from the data stored with
+accounts it does not contribute to the state hash computation.
+
+The entire mapping is stored "flat", meaning that any updates to the map cause
+the entire map to be written anew. This is alleviated by the fact that the size
+of the map is limited by the number of accounts with currently active releases,
+and the map is only updated upon an active release.
+
+### Transaction outcomes
+
+For each block we store the outcomes of transactions in that block. The outcome
+includes information about the order in the block, whether it succeeded or
+failed, the energy and GTU cost of transactions, and any events that resulted
+from the transaction. Events are transaction-type specific, and include things
+like GTU transfer, and smart contract updates.
+
+Transaction outcomes are currently stored as a "flat" structure, as a serialized
+list of transaction outcomes. The outcomes for a block are never updated once
+they are created, so this is not too wasteful, the only downside is that it
+prevents individual indexing, i.e., looking up the 5th transaction in the block.
+
+This structure does not contribute to the block state hash.
```

Comment:
It probably is worth mentioning that the transaction outcomes are hashed, and this hash is included in the block (along with the block state hash).